velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -1042,10 +1060,51 @@ void assertResults(
   }
 
   auto expectedRows = duckDbQueryRunner.execute(duckDbSql, resultType);
-  assertEqualResults(
-      expectedRows,
-      actualRows,
-      kDuckDbTimestampWarning + "\nDuckDB query: " + duckDbSql);
+
+  // For logical types like DECIMAL, we need to create the TypePtr
+  // from the plan provided for using it with variant::toJson().
```

Comment:
1) We need to ensure the `resultType` matches the type from the `results` vector.
2) We then simply pass the resultType.

Reply:
@majetideepak - In this case, we might need to do overload some of the methods and make it tidy.

Reply:
@majetideepak - There are cases where the `results` vector is empty, probably due to to no results available with the query but the resultType would by available from the plan. In such case, how can we ensure that type matches ?

Reply:
skip the type check if the results vector is empty. Later we have a row count check that will catch this.